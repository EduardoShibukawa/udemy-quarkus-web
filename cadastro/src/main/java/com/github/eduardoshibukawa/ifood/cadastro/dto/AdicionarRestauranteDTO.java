package com.github.eduardoshibukawa.ifood.cadastro.dto;

public class AdicionarRestauranteDTO {
    public String proprietario;

    public String cnpj;

    public String nomeFantasia;

    public LocalizacaoDTO localizacao;
}
