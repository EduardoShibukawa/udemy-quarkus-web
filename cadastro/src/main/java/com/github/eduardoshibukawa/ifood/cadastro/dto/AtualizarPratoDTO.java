package com.github.eduardoshibukawa.ifood.cadastro.dto;

import java.math.BigDecimal;

public class AtualizarPratoDTO {
    public BigDecimal preco;    
}
