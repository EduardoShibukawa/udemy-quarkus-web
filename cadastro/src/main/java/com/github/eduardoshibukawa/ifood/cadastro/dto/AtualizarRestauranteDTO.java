package com.github.eduardoshibukawa.ifood.cadastro.dto;

public class AtualizarRestauranteDTO {
    public String nomeFantasia;
}
