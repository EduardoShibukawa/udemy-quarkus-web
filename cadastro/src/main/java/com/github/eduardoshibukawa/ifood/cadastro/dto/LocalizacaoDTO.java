package com.github.eduardoshibukawa.ifood.cadastro.dto;

public class LocalizacaoDTO {
    
    public Double latitude;

    public Double longitude;
                    
}
