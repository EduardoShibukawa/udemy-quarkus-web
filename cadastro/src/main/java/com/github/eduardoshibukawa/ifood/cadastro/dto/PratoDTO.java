package com.github.eduardoshibukawa.ifood.cadastro.dto;

import java.math.BigDecimal;

public class PratoDTO {

    public Long id;

    public String nome;

    public String descricao;

    public BigDecimal preco;
}
