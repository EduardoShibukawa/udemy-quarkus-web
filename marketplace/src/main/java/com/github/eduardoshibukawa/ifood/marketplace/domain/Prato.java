package com.github.eduardoshibukawa.ifood.marketplace.domain;

import java.math.BigDecimal;
import java.util.stream.StreamSupport;

import com.github.eduardoshibukawa.ifood.marketplace.dto.PratoDTO;

import io.smallrye.mutiny.Multi;
import io.smallrye.mutiny.Uni;
import io.vertx.mutiny.pgclient.PgPool;
import io.vertx.mutiny.sqlclient.Row;
import io.vertx.mutiny.sqlclient.RowSet;
import io.vertx.mutiny.sqlclient.Tuple;

public class Prato {
    
    public Long id;
    
    public String nome;

    public String descricao;

    public Restaurante restaurante;

    public BigDecimal preco;

    public static Multi<PratoDTO> findAll(PgPool pgPool) {        
        Uni<RowSet<Row>> uniRowSet = pgPool.preparedQuery("select * from prato").execute();

        return transformUniToMulti(uniRowSet);
	}

    public static Multi<PratoDTO> findByRestaurante(PgPool pgPool, Long idRestaurante) {
        final String query = ""
            + "SELECT * "
            + "  FROM prato "
            + " WHERE prato.restaurante_id = $1 "
            + " ORDER BY nome ASC";        

        Uni<RowSet<Row>> uniRowSet 
            = pgPool.preparedQuery(query)
                    .execute(Tuple.of(idRestaurante));

        return transformUniToMulti(uniRowSet);
	}

    private static Multi<PratoDTO> transformUniToMulti(Uni<RowSet<Row>> uniRowSet) {
        return uniRowSet.onItem().transformToMulti(rowSet -> Multi.createFrom().items(
            () -> {
                return StreamSupport.stream(rowSet.spliterator() , false);
            }
        )).onItem().transform(PratoDTO::from);
    }
    
}
