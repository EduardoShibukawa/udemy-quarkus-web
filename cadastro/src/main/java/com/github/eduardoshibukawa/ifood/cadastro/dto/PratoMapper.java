package com.github.eduardoshibukawa.ifood.cadastro.dto;

import com.github.eduardoshibukawa.ifood.cadastro.domain.Prato;

import org.mapstruct.Mapper;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "cdi")
public interface PratoMapper {

    public Prato toPrato(AdicionarPratoDTO dto);

    public void toPrato(AtualizarPratoDTO dto, @MappingTarget Prato p);

    public PratoDTO toDTO(Prato p);
    
}
