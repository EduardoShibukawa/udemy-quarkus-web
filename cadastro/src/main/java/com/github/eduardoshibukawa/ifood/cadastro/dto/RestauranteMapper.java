package com.github.eduardoshibukawa.ifood.cadastro.dto;

import com.github.eduardoshibukawa.ifood.cadastro.domain.Restaurante;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "cdi")
public interface RestauranteMapper {

    @Mapping(target = "nome", source = "nomeFantasia")
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "dataCriacao", ignore = true)
    @Mapping(target = "dataAtualizacao", ignore = true)
    @Mapping(target = "localizacao.id", ignore = true)
    public Restaurante toRestaurante(AdicionarRestauranteDTO dto);

    @Mapping(target = "nome", source = "nomeFantasia")
    public void toRestaurante(AtualizarRestauranteDTO dto, @MappingTarget Restaurante r);

    @Mapping(target = "nomeFantasia", source = "nome")
    @Mapping(target = "dataCriacao", dateFormat = "dd/MM/yyyy HH:mm:ss")
    public RestauranteDTO toDTO(Restaurante r);
        
}
