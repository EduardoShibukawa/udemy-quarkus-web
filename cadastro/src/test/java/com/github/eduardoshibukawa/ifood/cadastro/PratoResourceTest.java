package com.github.eduardoshibukawa.ifood.cadastro;

import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.jupiter.api.Test;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;

import static io.restassured.RestAssured.given;

import java.math.BigDecimal;
import java.util.Optional;

import javax.ws.rs.core.Response.Status;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.eduardoshibukawa.ifood.cadastro.configuration.CadastroTestLifecycle;
import com.github.eduardoshibukawa.ifood.cadastro.domain.Prato;
import com.github.eduardoshibukawa.ifood.cadastro.dto.AdicionarPratoDTO;
import com.github.eduardoshibukawa.ifood.cadastro.dto.AtualizarPratoDTO;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycle.class)
public class PratoResourceTest {

    private static final String PRATOS_ENDPOINT = "/restaurantes/{idRestaurante}/pratos";
    private static final String PRATOS_CENARIO_1_YML = "pratos-cenario-1.yml";

    @Test
    @DataSet(PRATOS_CENARIO_1_YML)
    public void testGet() {
        final String resultado = given()
            .with().pathParam("idRestaurante", 123L)
            .when().get(PRATOS_ENDPOINT)
            .then()
            .statusCode(200)
            .extract().asString();

        Approvals.verifyAsJson(resultado);
    }

    @Test
    @DataSet(PRATOS_CENARIO_1_YML)
    public void testPost() {
        final AdicionarPratoDTO dto = new AdicionarPratoDTO();

        dto.preco = BigDecimal.valueOf(10d);
        dto.nome = "Saciroto";

        givenJsonContentWithIdRestauranteParam(123L)
            .body(dto)
            .when().post(PRATOS_ENDPOINT)
            .then()
            .statusCode(Status.CREATED.getStatusCode());

        final Long quantidadeEsperada = 2L;
        final Long quantidadeAtual = Prato.count();

        Assert.assertEquals(quantidadeEsperada, quantidadeAtual);
    }

    @Test
    @DataSet(PRATOS_CENARIO_1_YML)
    public void testPut() {
        final Long id = 321L;
        final AtualizarPratoDTO dto = new AtualizarPratoDTO();

        dto.preco = BigDecimal.valueOf(5d);

        givenJsonContentWithIdRestauranteParam(123L)
            .with().pathParam("id", id)
            .body(dto)
            .when().put(PRATOS_ENDPOINT + "/{id}")
            .then()
            .statusCode(Status.NO_CONTENT.getStatusCode());

        final Prato atual = Prato.findById(id);

        Assert.assertTrue("Preço deve ser atualizado após PUT", dto.preco.compareTo(atual.preco) == 0);        
    }

    @Test
    @DataSet(PRATOS_CENARIO_1_YML)
    public void testDelete() {
        final Long id = 321L;

        givenJsonContentWithIdRestauranteParam(123L)
            .with().pathParam("id", id)
            .when().delete(PRATOS_ENDPOINT + "/{id}")
            .then()
            .statusCode(Status.NO_CONTENT.getStatusCode());

        final Optional<Prato> op = Prato.findByIdOptional(id);

        Assert.assertTrue("Não deve existir prato", op.isEmpty());        
    }

    private RequestSpecification givenJsonContentWithIdRestauranteParam(Long idRestaurante) {
        return given()
            .contentType(ContentType.JSON)
            .with().pathParam("idRestaurante", idRestaurante);
    }        
}
