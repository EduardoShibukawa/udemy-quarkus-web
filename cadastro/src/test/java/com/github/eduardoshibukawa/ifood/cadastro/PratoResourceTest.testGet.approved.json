"[{\"descricao\":\"Lanche com ovo, bacon e hamburguer\",\"id\":321,\"nome\":\"Saci\",\"preco\":20.00}]"